### ISSUE TITLE
Better handling of preprocess errors

### ISSUE THREAD

**Issue by benmccann** [sveltejs/vite-plugin-svelte#259]
### Describe the problem

I was getting the error message:

>[vite-plugin-svelte]  Cannot read property 'split' of undefined
file: src/routes/__layout.svelte
 Cannot read property 'split' of undefined

This was impossible to track down.

This turned out to be a bug in `svelte-preprocess` that appears to have been recently fixed. It happens if you have `svelte-preprocess` 4.9.0 with the following code:
```
<script lang="ts" context="module">
</script>
<script>
</script>
```

Here's a commit where it was happening: https://github.com/svelte-society/sveltesociety.dev/pull/222/commits/78cdfa1dd453f8f0187ecd64fc3237e51ed44e89. You can reproduce with `npm run build`

### Describe the proposed solution

It would be really nice to get a stack trace. I'm not sure why that was being swallowed and I was just getting the error message

### Alternatives considered

We could put a try / catch around the call to preprocess to at least make it clear that's where it's coming from. `svelte` core could also do some type of reporting when a preprocessor fails. Neither solution would be as helpful as a stack trace though

### Importance

would make my life easier

**Comment by dominikg** [comment]
We missed this when we implemented https://github.com/sveltejs/vite-plugin-svelte/issues/218

I'll look into converting preprocessor errors in the same way. Let's hope there is frame info available

**Comment by benmccann** [comment]
There wouldn't be a frame in this particular case because that's something the Svelte compiler has to explicitly add and this was an unexpected error from a bug in svelte-preprocess. Maybe if there was a syntax error or something though

**Comment by dominikg** [comment]
fixed in [1.0.0-next.36](https://github.com/sveltejs/vite-plugin-svelte/blob/main/packages/vite-plugin-svelte/CHANGELOG.md#100-next36)
